aws/data_source_aws_organizations_organizational_unit.go

```diff
@@ -0,0 +1,76 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/organizations"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceAwsOrganizationsOrganizationalUnit() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsOrganizationsOrganizationalUnitRead,
+
+		Schema: map[string]*schema.Schema{
+			"parent_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"children": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"arn": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceAwsOrganizationsOrganizationalUnitRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).organizationsconn
+
+	parent_id := d.Get("parent_id").(string)
+	d.SetId(resource.UniqueId())
+
+	params := &organizations.ListOrganizationalUnitsForParentInput{
+		ParentId: aws.String(parent_id),
+	}
+
+	var children []*organizations.OrganizationalUnit
+	for {
+		ous, err := conn.ListOrganizationalUnitsForParent(params)
+
+		if err != nil {
+			return fmt.Errorf("Error listing organizational units for parent: %s", err)
+		}
+
+		children = append(children, ous.OrganizationalUnits...)
+
+		if ous.NextToken == nil {
+			break
+		}
+		params.NextToken = ous.NextToken
+	}
```

Comment:
The AWS Go SDK provides a paginated function, [`ListOrganizationalUnitsForParentPages`](https://docs.aws.amazon.com/sdk-for-go/api/service/organizations/#Organizations.ListOrganizationalUnitsForParentPages), which can be used to simplify this logic:

```go
err := conn.ListOrganizationalUnitsForParentPages(params,
    func(page *organizations.ListOrganizationalUnitsForParentOutput, lastPage bool) bool {
  children = append(children, page.OrganizationalUnits...)

  return !lastPage
})

if err != nil {
  return fmt.Errorf("error listing Organizations Organization Units for parent (%s): %s", parent_id, err)
}
```